var/spack/repos/builtin/packages/duckdb/package.py

```diff
@@ -16,8 +16,65 @@ class Duckdb(CMakePackage):
 
     license("MIT")
     maintainers("glentner")
+    maintainers("teaguesterling")
 
     version("master", branch="master")
+    version("0.10.0", sha256="5a925b8607d00a97c1a3ffe6df05c0a62a4df063abd022ada82ac1e917792013")
     version("0.9.2", sha256="afff7bd925a98dc2af4039b8ab2159b0705cbf5e0ee05d97f7bb8dce5f880dc2")
+    version("0.9.1", sha256="37a43188d9354ce3ca101b2b118d867f5f76d04c3b83c09d86fd7508351a631b")
+    version("0.9.0", sha256="3dbf3326a831bf0797591572440e81a3d6d668f8e33a25ce04efae19afc3a23d")
     version("0.8.1", sha256="a0674f7e320dc7ebcf51990d7fc1c0e7f7b2c335c08f5953702b5285e6c30694")
+    version("0.8.0", sha256="df3b8e0b72bce38914f0fb1cd02235d8b616df9209beb14beb06bfbcaaf2e97f")
     version("0.7.1", sha256="67f840f861e5ffbe137d65a8543642d016f900b89dd035492d562ad11acf0e1e")
+
+    depends_on("python@3:")
+    depends_on("cmake")
+
+    variant("autocomplete", default=True, description="Include autocomplete for CLI in build")
+    variant("icu", default=True, description="Compile with bundled ICU library")
+    variant("json", default=True, description="Include JSON support in build")
+    variant("httpfs", default=True, description="Include HTTPFS (& S3) support in build")
+    variant("excel", default=True, description="Include Excel formatting extension in build")
+    variant("openssl", default=True, description="Compile with bundled OpenSSl library")
+    variant("odbc", default=False, description="Build with ODBC driver (may not work)")
+    variant("jdbc", default=False, description="Build JDBC driver (may not work)")
+    variant("inet", default=True, description="Include INET (ip address) support in build")
+    variant("fts", default=True, description="Include FTS (full text search) support in build")
+    variant("parquet", default=True, description="Include parquent support in build")
+
+    def edit(self, spec, prefix):
+        extensions = [
+            "autocomplete",
+            "icu",
+            "json",
+            "httpfs",
+            "excel",
+            "openssl",
+            "odbc",
+            "jdbc",
+            "fts",
+            "parquet",
+        ]
+        for flag in extensions:
+            make_flag = "BUILD_" + flag.upper()
+            if "+" + flag in spec:
+                env[make_flag] = "1"
+            elif "~" + flag in spec:
+                env[make_flag] = "0"
```

Comment:
The duckdb docs recommend building with ninja.
Not sure if it makes much of a difference for spack, but it might work if we add a dependency for ninja and then add the "GEN=ninja" environment variable here.

Reply:
Works with or without Ninja. Added a (default=True) variant to use ninja as proposed here and set up the necessary dependencies.